src/clients/discord/discord.voice.service.ts

```diff
@@ -257,30 +278,54 @@ export class DiscordVoiceService implements OnModuleDestroy {
     return true;
   }
 
-  disconnect(): TryResult<
-    string | MessagePayload | InteractionEditReplyOptions
-  > {
-    if (this.voiceConnection === undefined) {
+  disconnect(): TryResult<string | MessagePayload | InteractionEditReplyOptions> {
+    // 🧩 1️⃣ Early exit if already disconnected
```

Comment:
I assume you worked on this using AI?
I don't mind the comments here, as they give useful information on what is going on.
But please stay careful when using AI, it often hallucinates.